presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkMapFlatWriter.java

```diff
@@ -123,64 +129,125 @@ private void doWrite(BenchmarkData data)
-
-        @Param("bigint")
-        private String valueType = "bigint";
+        @Param({"map<bigint,bigint>", "map<bigint,array<bigint>>"})
+        private String typeSignature = "map<bigint,bigint>";
 
-        @Param("5000")
-        private int distinctKeyCount = 5000;
+        // total number of keys
+        @Param("500")
+        private int distinctKeyCount = 500;
 
+        // how many positions each block will have
         @Param("50000")
         private int positionsPerBlock = 50000;
 
-        @Param("500")
-        private int maxKeysPerEntry = 500;
-
+        // how many blocks to create
         @Param("5")
         private int blockCount = 5;
 
+        // number of values for the complex value types
+        @Param("10")
+        private int childElementValueCount = 10;
+
+        // Weights for key frequencies bucketed by 10%.
+        // It has 11 elements. The element position is the frequency bucket.
+        // The value of the element is the weight indicating how many keys should
+        // fall into this frequency bucket. All weights will be normalized before usage.
+        // The last element is the weight for keys with 100% frequency.
+        //
+        // Illustration:
+        // [this many keys (after weights normalization) come with frequency <10%, ... with frequency <20%, ..., ...with frequency 100%]
+        //
+        // For example:
+        // [1, 1, 0,0,0,0,0,0,0,0, 1] means that 1/3 of keys would come with 10% frequency, another 1/3 with 20% frequency, and the last 1/3 with 100%.
+        @Param("12, 2, 5, 3, 4, 5, 2, 10, 7, 45, 0")
+        private String frequencyHistogram = "12, 2, 5, 3, 4, 5, 2, 10, 7, 45, 0";
+
         private List<Block> blocks;
 
         @Setup
         public void setup()
                 throws Exception
         {
-            FunctionAndTypeManager testFunctionAndTypeManager = createTestFunctionAndTypeManager();
-            Type keyType = testFunctionAndTypeManager.getType(parseTypeSignature(this.keyType));
-            Type valueType = testFunctionAndTypeManager.getType(parseTypeSignature(this.valueType));
+            Random random = new Random(SEED);
+
+            FunctionAndTypeManager typeManager = createTestFunctionAndTypeManager();
+            this.type = (MapType) typeManager.getType(parseTypeSignature(typeSignature));
+            Type keyType = type.getKeyType();
+            Type valueType = type.getValueType();
 
-            this.type = mapType(keyType, valueType);
             this.temporaryDirectory = createTempDir();
             this.orcFile = new File(temporaryDirectory, randomUUID().toString());
+            this.orcFile.deleteOnExit();
 
             ImmutableList.Builder<Block> blocks = ImmutableList.builder();
 
             List<Long> keys = new ArrayList<>();
             for (int i = 0; i < distinctKeyCount; i++) {
                 keys.add((long) i);
             }
+            Collections.shuffle(keys, random);
```

Comment:
Initialize the keys arrayList with Size. I also recommend doing n random numbers. You can do n random numbers like this.

Set<Long> keySet = new HashSet<Long>();
while (keySet.size() < distinctKeyCount) {
   keySet.add(rand.nextLong());
}

convert this set to list again.